runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -1571,6 +1619,10 @@ def HandleEvaluate(msg: string)
     ignoreEvalError = true
     SendEval($'*{evalexpr}')
   else
+    if evalInPopup
+      Popup_show(evalExprResult)
+      evalInPopup = false
+    endif
```

Comment:
Why not moving the `Popup_show()` call a bit before, like for example: 

```diff
index 667da34a2..03e401fbf 100644
--- a/runtime/pack/dist/opt/termdebug/plugin/termdebug.vim
+++ b/runtime/pack/dist/opt/termdebug/plugin/termdebug.vim
@@ -1609,6 +1609,8 @@ def HandleEvaluate(msg: string)
     endif
     if evalFromBalloonExpr
       Balloon_show(evalExprResult)
+    elseif evalInPopup
+      Popup_show(evalExprResult)
     endif
   else
     echomsg $'"{evalexpr}": {value}'
@@ -1619,10 +1621,7 @@ def HandleEvaluate(msg: string)
     ignoreEvalError = true
     SendEval($'*{evalexpr}')
   else
-    if evalInPopup
-      Popup_show(evalExprResult)
-      evalInPopup = false
-    endif
+    evalInPopup = false
     evalFromBalloonExpr = false
   endif
 enddef
 ``` 

Is there any drawback?

Reply:
I initially didn't actively close the popup window. So I moved the call further down to avoid multiple windows being opened in case a pointer was dereferenced (and SendEval was called a 2nd time).

In the current version, I close the last popup window anyway, so the call could be moved back up.

Reply:
I'm not that familiar with "balloon" windows, but it seems that the balloon opened with "balloon_show" will be closed automatically, while the "popup_atcursor" will only be closed when either closed manually or the cursor moves.

```bash
# This will create two overlapping windows which seem to spell "Hillo"
vim -u NONE -c 'call popup_atcursor("Hi", {}) | call popup_atcursor("Hello", {})'

# This on the other hand will only show the latest balloon. The first one seems to be closed automatically.
vim -u NONE -g -c 'call balloon_show("Hi") | call balloon_show("Hello")'
```

So, yes: When dereferencing a pointer expr, `balloon_show` is called twice, but it never caused any visual issue.

Treating both in the same way, seems reasonable. What do you think? Should I move the popup window code up or pull the balloon code down?

Reply:
I'd also recommend to move it up, if that doesn't cause issues.

Reply:
Done with new commits.

Reply:
One more thing on this point and then we should be done: if moving both balloon and popup up cause an unnecessary double call plus a non-optimal visual experience in the case of popup whereas moving both down cause a single call and the same optimal visual experience, then perhaps it's better to move both down instead? Is that the case? 

I would like to keep them together for making the code easier to maintain.

Reply:
I've updated this again and moved both down to avoid unnecessary calls.

Reply:
Perfect! Thanks!